File: python3/3239.py
class Solution:
    def minFlips(self, grid: List[List[int]]) -> int:
        m = len(grid)
        n = len(grid[0])
        rc = 0
        cc = 0
        for i in grid:
            s = 0
            e = n - 1
            while s < e:
                if i[s] != i[e]:
                    rc += 1
                s += 1
                e -= 1
        for i in range(n):
            s = 0
            e = m - 1
            while s < e:
                if grid[s][i] != grid[e][i]:
                    cc += 1
                s += 1
                e -= 1
        return min(rc, cc)

File: java/3239.java
class Solution {
    public int minFlips(int[][] grid) {
        int m = grid.length, n = grid[0].length;
        int rc = 0, cc = 0;
        for (int i = 0; i < m; i++) {
            int s = 0, e = n - 1;
            while (s < e)
                rc += grid[i][s++] ^ grid[i][e--];
        }
        for (int i = 0; i < n; i++) {
            int s = 0, e = m - 1;
            while (s < e)
                cc += grid[s++][i] ^ grid[e--][i];
        }
        return ((rc < cc) ? rc : cc);
    }
}
